### PR TITLE
improve tests

### PR DESCRIPTION
**Pull request by dipeshkaphle**

### CODE REVIEW

File: test/incr_test.ml [hunk 1]
```diff
@@ -270,6 +270,247 @@ let bind_test () =
...
+            Int.add x y)
+          (sum_range ~lo ~hi:mid xs) (sum_range ~lo:mid ~hi xs)
+    in
+    let arr = Array.of_list [1; 2; 3; 4; 5; 6; 7; 8; 9; 10] in
+    let var_arr = Array.map Var.create arr in
+    let t_arr = Array.map Var.watch var_arr in
+    let sum = sum_range ~lo:0 ~hi:(Array.length t_arr) t_arr in
+    let executor =
+      {
+        run = (fun f -> f ());
```
Comment (polytypic) [inline test/incr_test.ml:296]: BTW, I often format records, especially records with functions, such that I first bind the values with the same names as the record fields and then use the record punning notation:

```ocaml
let executor =
  let run f = f ()
  and par_do l r =
    let lres = l () in
    (lres, r ())
  in
  { run; par_do }
```

This often formats to fewer lines and the syntax for functions can be nicer.

File: test/incr_test.ml [hunk 2]
```diff
@@ -270,6 +270,247 @@ let bind_test () =
...
+      end
+      else
+        let mid = lo + ((hi - lo) asr 1) in
+        map2
+          ~fn:(fun x y ->
+            incr ops;
+            Int.add x y)
+          (sum_range ~lo ~hi:mid xs) (sum_range ~lo:mid ~hi xs)
+    in
+    let arr = Array.of_list [1; 2; 3; 4; 5; 6; 7; 8; 9; 10] in
```
Comment (polytypic) [inline test/incr_test.ml:290]: `Array.init` tends to be convenient for these.

```ocaml
# Array.init 10 ((+) 1) ;;
- : int array = [|1; 2; 3; 4; 5; 6; 7; 8; 9; 10|]
```

**Comment by dipeshkaphle** [comment]
I have made changes addressing the comments

File: test/incr_test.ml [hunk 3]
```diff
@@ -13,205 +13,124 @@ let live_words () =
 let does_raise f = match f () with exception _ -> true | _ -> false
```
Comment (polytypic) [inline test/incr_test.ml:13]: How about `Alcotest.check_raises`?

Comment (dipeshkaphle) [reply test/incr_test.ml:13]: I'll change it to use that

File: test/incr_test.ml [hunk 4]
```diff
@@ -13,205 +13,124 @@ let live_words () =
...
-        run = seq_executor.run;
-        par_do =
-          (fun l r ->
-            incr par_do_call_count;
-            seq_executor.par_do l r);
-      }
+    let run f = T.run pool f in
+
+    let par_do l r =
+      incr par_do_call_count;
```
[the remark this answers is not in the document]
Comment (dipeshkaphle) [reply test/incr_test.ml:115]: Yeah I'll add why it's not necessary.

**Comment by polytypic** [comment]
As a general comment it could be helpful to have a short comment in each test about what the test is about: "This tests that ... by doing ...".

**Comment by dipeshkaphle** [comment]
That makes sense. I'll add a short comment describing what I'm trying to test exactly

**Review by polytypic — APPROVED** [review]

**Pull request merged**